designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,285 @@
+aligning with their domain of responsibility.
+- **Flexible Network Policy Usage**: Allow users to opt-out of the default-enabled usage of [Network Policies][k8s-doc-networkpolicies]
+if they prefer another solution, plan to deploy their solution with a vendor, or use a CNI that does not support NetworkPolicies.
+
+### Non-Goals
+
+- **Replicate kube-rbac-proxy Features or Protection Level**: It is not a goal to provide exactly the same features
+or layer of protection as [kube-rbac-proxy][kube-rbac-proxy]. Since [Network Policies][k8s-doc-networkpolicies]operate differently
+and do not offer the same kind of functionality as [kube-rbac-proxy][kube-rbac-proxy], achieving identical protection levels through
+[Network Policies][k8s-doc-networkpolicies]alone is not feasible.
+
+However, incorporating NetworkPolicies, cert-manager, and/or the features introduced
+in the [controller-runtime pull request #2407][cr-pr] we are mainly addressing the security concerns that
+kube-rbac-proxy handles.
+
+## Proposal
+
+### Phase 1: Transition to NetworkPolicies
+
+The immediate action outlined in this proposal is the replacement of [kube-rbac-proxy][kube-rbac-proxy]
+with Kubernetes API NetworkPolicies.
+
+### Phase 2: Add Cert-Manager as Optional option to be used with metrics
+
+Looking beyond the initial phase, this proposal envisions integrating cert-manager for TLS certificate management
+and exploring synergies with new features in Controller Runtime, as demonstrated in [PR #2407](https://github.com/kubernetes-sigs/controller-runtime/pull/2407).
+
+These enhancements would introduce encrypted communication for metrics endpoints and potentially incorporate authentication mechanisms,
+significantly elevating the security model employed by projects scaffolded by Kubebuilder.
+
+- **cert-manager**: Automates the management and issuance of TLS certificates, facilitating encrypted communication and, when configured with mTLS, adding a layer of authentication.
+  Currently, we leverage on cert-manager when webhooks are scaffold. So, the proposal idea would be to allow users enable the cert-manager for the metrics such as it is provided
+  and required for webhook feature. However, it MUST be optional. One of the goals of Kubebuilder is make it easier for new users, therefore new users should
+  not need to deal with cert-manager by default or have the need to install it to just quick start.
+
+### Phase 3: When Controller-Runtime feature be enhanced
+
+After we have the [issue](https://github.com/kubernetes-sigs/controller-runtime/issues/2781)
+addressed we plan to use it to protect the endpoint. See that would mean ensure
+that we are either `handle authentication (authn), authorization (authz)`.
+Examples of its implementation can be found [here](https://github.com/sbueringer/controller-runtime-tests/blob/master/metrics-auth/withmetricsauth/main.go#L10-L14)
```

Comment:
I would prefer linking to the CAPI implementation instead: https://github.com/kubernetes-sigs/cluster-api/blob/v1.6.3/util/flags/diagnostics.go#L79-L82

Reply:
(also the second link below)

Reply:
```suggestion
Examples of its implementation can be found [here](https://github.com/kubernetes-sigs/cluster-api/blob/v1.6.3/util/flags/diagnostics.go#L79-L82) 
```